src/scripts/dashboardpage.js

```diff
@@ -1,447 +1,928 @@
-                name: globalize.translate("ViewPlaybackInfo"),
-                id: "transcodinginfo"
-            }), actionsheet.show({
+
+            if (session.ServerId && session.DeviceId !== connectionManager.deviceId()) {
+                menuItems.push({
+                    name: globalize.translate("SendMessage"),
+                    id: "sendmessage"
+                });
+            }
+
+            if (session.TranscodingInfo && session.TranscodingInfo.TranscodeReasons && session.TranscodingInfo.TranscodeReasons.length) {
+                menuItems.push({
+                    name: globalize.translate("ViewPlaybackInfo"),
+                    id: "transcodinginfo"
+                });
+            }
+
+            return actionsheet.show({
                 items: menuItems,
                 positionTo: btn
-            }).then(function(id) {
+            }).then(function (id) {
                 switch (id) {
                     case "sendmessage":
                         showSendMessageForm(btn, session);
                         break;
+
                     case "transcodinginfo":
-                        showPlaybackInfo(btn, session)
+                        showPlaybackInfo(btn, session);
                 }
-            })
-        })
+            });
+        });
     }
 
-    function onActiveDevicesClick(e) {
-        var btn = dom.parentWithClass(e.target, "sessionCardButton");
+    function onActiveDevicesClick(e__r) {
+        var btn = dom.parentWithClass(e__r.target, "sessionCardButton");
+
         if (btn) {
             var card = dom.parentWithClass(btn, "card");
+
             if (card) {
-                var sessionId = card.id,
-                    session = (DashboardPage.sessionsList || []).filter(function(s) {
-                        return "session" + s.Id === sessionId
-                    })[0];
-                session && (btn.classList.contains("btnCardOptions") ? showOptionsMenu(btn, session) : btn.classList.contains("btnSessionInfo") ? showPlaybackInfo(btn, session) : btn.classList.contains("btnSessionSendMessage") ? showSendMessageForm(btn, session) : btn.classList.contains("btnSessionStop") ? connectionManager.getApiClient(session.ServerId).sendPlayStateCommand(session.Id, "Stop") : btn.classList.contains("btnSessionPlayPause") && session.PlayState && connectionManager.getApiClient(session.ServerId).sendPlayStateCommand(session.Id, "PlayPause"))
+                var sessionId = card.id;
+                var session = (DashboardPage.sessionsList || []).filter(function (s__t) {
+                    return "session" + s__t.Id === sessionId;
+                })[0];
+
+                if (session) {
+                    if (btn.classList.contains("btnCardOptions")) {
+                        showOptionsMenu(btn, session);
+                    } else {
```

Comment:
couldn't you just combine this in a form of
```js
if (session) {
   // stuff
} else if (other check) {
   // other stuff
} else if (booo) {
   // bar baz
}
```
instead of nesting `else`s?